api/src/Responder/Responder/ApiResponder.php

```diff
@@ -137,16 +143,65 @@ protected function collectionWithStatusCode($items, $code = 200)
     }
 
     /**
-     * TODO: Implement paginator() method.
-     * Bind a paginator to a transformer and start building a response.
+     * Build paginated response.
      *
-     * @param Paginator $paginator
+     * @param Collection|array $items
+     * @param null|int $offset
+     * @param null|int $totalCount
      * @param array $parameters
      *
      * @return Response
      */
-    public function paginator(Paginator $paginator, array $parameters = [])
+    public function paginatedCollection($items, $offset = null, $totalCount = null, array $parameters = [])
+    {
+        $itemCount = count($items);
+        $this->validateRange($itemCount);
+
+        $response = $this->getResponse();
+        $response->headers->set('Accept-Ranges', 'entities');
+        $response->headers->set('Content-Type', 'application/json');
+        $response->setStatusCode(206);
+
+        $rangeHeader = $this->prepareRangeHeader($itemCount, $offset, $totalCount);
+        $response->headers->set('Content-Range', $rangeHeader);
+
+        $response->setContent($this->encode($this->getTransformer()->transformCollection($items)));
+
+
+
```

Comment:
This function should be using chained methods of the responder. Actually this responder class should extend Response, so that it can be used directly in a chain.